kobo/client/__init__.py

```diff
@@ -183,6 +183,23 @@ def __getattr__(self, name):
         except:
             raise AttributeError("'%s' object has no attribute '%s'" % (self.__class__.__name__, name))
 
+    @property
+    def is_logged_in(self):
+        return self._logged_in
+
+    def retry_login(self):
```

Comment:
Could we just name it `login` please?  The `retry_` prefix feels weird.  Consider that any method which can be called more than once can be called in order to retry something.

Also, what do you think of changing it to raise on failure instead of return a bool?  The problem with returning False is of course that people can forget to check the returned value.  Also, all conscientious callers are going to have to check the result and raise anyway - may as well do it just once in a central place.

Reply:
Sure I'll rename it. Im not married to `retry_login`, I just called it that since hub attempts auth once for the caller.

I'm a fan of raising on failure, I will make that change as well. I'll look at the existing code first to see if something appropriate already exists, but if not are you opposed to a new exception type like `KoboHubLoginFailure`?

Reply:
Yeah, a new exception type would be fine.

Reply:
Found that there is already an `AuthenticationError` class, so I used that instead of adding a new type.